redisinsight/api/test/api/database-analysis/POST-databases-id-analysis.test.ts

```diff
@@ -134,14 +135,80 @@ describe('POST /databases/:instanceId/analysis', () => {
         expect(body.topKeysLength[0].length).to.gt(0);
 
         expect(body.expirationGroups.length).to.eq(8);
-        for(let i = 1; i < 8; i++) {
+        for (let i = 1; i < 8; i++) {
           expect(body.expirationGroups[i].label).to.be.a('string');
           expect(body.expirationGroups[i].total).to.eq(0);
           expect(body.expirationGroups[i].threshold).to.gt(0);
         }
         expect(body.expirationGroups[0].label).to.eq('No Expiry');
         expect(body.expirationGroups[0].total).to.gt(0);
         expect(body.expirationGroups[0].threshold).to.eq(0);
+        expect(body.recommendations.length).to.eq(0);
+      },
+      after: async () => {
+        expect(await repository.count()).to.eq(5);
+      }
+    },
+    {
+      name: 'Should create new database analysis with useSmallerKeys recommendation',
+      data: {
+        delimiter: '-',
+      },
+      statusCode: 201,
+      responseSchema,
+      before: async () => {
+        const KEYS_NUMBER = 1_000_001
+        await rte.data.generateNKeys(KEYS_NUMBER, false);
+      },
+      checkFn: async ({ body }) => {
+        expect(body.totalKeys.total).to.gt(0);
+        expect(body.totalMemory.total).to.gt(0);
+        expect(body.topKeysNsp.length).to.gt(0);
+        expect(body.topMemoryNsp.length).to.gt(0);
+        expect(body.topKeysLength.length).to.gt(0);
+        expect(body.topKeysMemory.length).to.gt(0);
+        expect(body.recommendations).to.deep.eq([{ name: 'useSmallerKeys'}]);
```

Comment:
Better to move to constants the same way you did for **TEST_DATABASE_ANALYSIS_RECOMMENDATION_1**. But not necessary.